dir/path.go

```diff
@@ -1,64 +1,126 @@
 package dir
 
 import (
+	"errors"
+	"fmt"
+	"io/fs"
+
 	"github.com/opencontainers/go-digest"
 )
 
 const (
+	// CertificateExtension defines the extension of the certificate files
+	CertificateExtension = ".crt"
+
+	// ConfigFile is the name of config file
+	ConfigFile = "config.json"
+
+	// KeyExtension defines the extension of the key files
+	KeyExtension = ".key"
+
+	// LocalKeysDir is the directory name for local key store
+	LocalKeysDir = "localkeys"
+
 	// SignatureExtension defines the extension of the signature files
 	SignatureExtension = ".sig"
+
+	// SignatureStoreDirName is the name of the signature store directory
+	SignatureStoreDirName = "signatures"
+
+	// SigningKeysFile is the file name of signing key info
+	SigningKeysFile = "signingkeys.json"
+
+	// TrustPolicyFile is the file name of trust policy info
+	TrustPolicyFile = "trustpolicy.json"
+
+	// TrustStoreDir is the directory name of trust store
+	TrustStoreDir = "truststore"
 )
 
 // PathManager contains the union directory file system and methods
 // to access paths of notation
 type PathManager struct {
-	ConfigFS  UnionDirFS
-	CacheFS   UnionDirFS
-	LibexecFS UnionDirFS
+	ConfigFS     UnionDirFS
+	CacheFS      UnionDirFS
+	LibexecFS    UnionDirFS
+	UserConfigFS UnionDirFS
+}
+
+func checkError(err error) {
+	// if path does not exist, the path can be used to create file
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		panic(err)
```

Comment:
Do we really need to panic here? Returning errors seems to be a better option for Go libraries per https://stackoverflow.com/a/35413011